File: Native_Kotlin/app/src/main/java/com/example/activityautostart/activities/SelectorActivity.kt
package com.example.activityautostart.activities

import android.app.Activity
import android.content.Context
import android.graphics.drawable.GradientDrawable
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.text.Editable
import com.example.activityautostart.data.DataRepository
import com.example.activityautostart.data.TimingRepository
import com.example.activityautostart.selector.SelectorAdapter
import com.example.activityautostart.selector.SelectorListController
import com.example.movch.activityautostart.R
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_selector.*

class SelectorActivity : AppCompatActivity() {

    private val preferencesName = "SETTINGS_DEFAULT"
    private val dataKey = "data"
    private val intervalKey = "interval"
    private val showingKey = "showing"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_selector)
    }

    override fun onResume() {
        super.onResume()

        getFromShared()

        val controller = SelectorListController(selectorRecycler)
        selectorRecycler.adapter = SelectorAdapter(controller)
        selectorRecycler.layoutManager = LinearLayoutManager(this, 1, false)

        controller.init()

        this.setResult(Activity.RESULT_OK, intent)
        initButton()
    }

    private fun initButton() {
        selectorBtnConfirm.setOnClickListener {
            setIntoShared()
            this.onBackPressed()
        }

//        runOnUiThread {
//            mainBtnStart.background.mutate()
//            val drawer = mainBtnStart.background as GradientDrawable
//            drawer.setStroke(2, getColor(R.color.textColorMain))
//        }
    }

    private fun setIntoShared() {
        val editor = getSharedPreferences(preferencesName, Context.MODE_PRIVATE).edit()
        val set = HashSet<String>()
        for (data in DataRepository.getCheckedData())
            set.add(data.id)

        TimingRepository.setInterval(selectorInterval.text.toString().toInt())
        TimingRepository.setShowingTime(selectorTime.text.toString().toInt())

        editor.putStringSet(dataKey, set)
        editor.putInt(intervalKey, selectorInterval.text.toString().toInt())
        editor.putInt(showingKey, selectorTime.text.toString().toInt())
        editor.apply()
    }

    private fun getFromShared() {
        val preferences = getSharedPreferences(preferencesName, Context.MODE_PRIVATE)
        val dataIds = preferences.getStringSet(dataKey, HashSet<String>())
        val interval = preferences.getInt(intervalKey, 240)
        val showingTime = preferences.getInt(showingKey, 30)

        DataRepository.setCheckedData(dataIds)
        selectorInterval.text = Editable.Factory.getInstance().newEditable(interval.toString())
        selectorTime.text = Editable.Factory.getInstance().newEditable(showingTime.toString())
    }
}

File: Xamarin_CSharp/ActivityAutostart/Activities/SelectorActivity.cs
using System.Collections.Generic;
using System.Linq;
using ActivityAutostart.Views;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Support.V7.App;
using Android.Widget;
using Portable.Controllers.Selector;
using Portable.Controllers.Selector.Interfaces;
using Portable.Repositories;

namespace ActivityAutostart.Activities
{
    [Activity(Label = "SelectorActivity")]
    public class SelectorActivity : AppCompatActivity
    {

        private const string PreferencesName = "SETTINGS_DEFAULT";
        private const string DataKey = "data";
        private const string IntervalKey = "interval";
        private const string ShowingKey = "showing";
        private const string SkipKey = "skip";

        private bool _wasShowed = false;

        private ISelectorController _controller;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_selector);
        }

        protected override void OnResume()
        {
            base.OnResume();

            GetFromShared();
            InitController();
            _controller.Subscribe();
            _controller.OnReturn += SetIntoShared;

            if (_wasShowed)
            {
                var button = FindViewById<Button>(Resource.Id.selectorBtnConfirm);
                button.PerformClick();
            }
        }

        protected override void OnPause()
        {
            base.OnPause();

            _controller.Unsubscribe();
            _controller.OnReturn -= SetIntoShared;
        }

        private void GetFromShared()
        {
            var preferences = GetSharedPreferences(PreferencesName, FileCreationMode.Private);
            var dataIds = preferences.GetStringSet(DataKey, new List<string>());
            var interval = preferences.GetInt(IntervalKey, 240);
            var showingTime = preferences.GetInt(ShowingKey, 30);
            _wasShowed = preferences.GetBoolean(SkipKey, false);

            DataRepository.GetInstance().SetCheckedData(dataIds.ToList());

            var selectorInterval = FindViewById<EditText>(Resource.Id.selectorInterval);
            var selectorTime = FindViewById<EditText>(Resource.Id.selectorTime);

            selectorInterval.Text = interval.ToString();
            selectorTime.Text = showingTime.ToString();
        }

        private void InitController()
        {
            var view = FindViewById<SelectorView>(Resource.Id.selectorView);

            _controller = new SelectorController(view);
            _controller.Init();
        }

        private void SetIntoShared()
        {
            var editor = GetSharedPreferences(PreferencesName, FileCreationMode.Private).Edit();
            var list = new List<string>();
            foreach (var data in DataRepository.GetInstance().CheckedData)
            {
                list.Add(data.Id);
            }

            var selectorInterval = FindViewById<EditText>(Resource.Id.selectorInterval);
            var selectorTime = FindViewById<EditText>(Resource.Id.selectorTime);
            
            int.TryParse(selectorInterval.Text, out int interval);
            int.TryParse(selectorTime.Text, out int showingTime);

            TimingRepository.GetInstance().Interval = interval;
            TimingRepository.GetInstance().ShowingTime = showingTime;

            editor.PutStringSet(DataKey, list);
            editor.PutInt(IntervalKey, interval);
            editor.PutInt(ShowingKey, showingTime);
            editor.PutBoolean(SkipKey, true);
            editor.Apply();

            RunOnUiThread(OnBackPressed);
        }
    }
}